types/node/child_process.d.ts

```diff
@@ -4,17 +4,18 @@ declare module "child_process" {
     import * as net from "net";
 
     interface ChildProcess extends events.EventEmitter {
-        stdin: stream.Writable;
-        stdout: stream.Readable;
-        stderr: stream.Readable;
-        stdio: [stream.Writable, stream.Readable, stream.Readable];
-        killed: boolean;
-        pid: number;
+        stdin: stream.Writable | null;
+        stdout: stream.Readable | null;
+        stderr: stream.Readable | null;
+        readonly channel: boolean;
```

Comment:
I don't think `channel` is a `boolean`. I player around and got `null` and `undefined` and according to the docs it's a pipe.

Reply:
Yea my bad, any idea what sort of `Pipe` it is, is that a Duplex stream or something? The NodeJS code is extremely hard to follow here.

Reply:
I think it's a `pipe_wrap` which is also used inside a `net.socket` if it is connected via a unix domain socket or windows pipe. Not sure if there exists a detailed doc or an existing type definition.

Reply:
We could create a bare-bone `Pipe` interface in the `stream` module, not sure if we are reaching too far into nodes core with that tho.

Reply:
Are you using this feature currently? Maybe we should wait until someone really needs this to have a real world test.

Reply:
Nah, but I just wanted to add it for the sake of completeness, I will make it an empty interface for now.